### PR TITLE
Added cask for Lego Mindstorms EV3 Home Edition IDE.

### PR DESCRIPTION
**Pull request by 2manyprojects2littletime**

### CODE REVIEW

File: Casks/lego-mindstorms-ev3.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cask :v1 => 'lego-mindstorms-ev3' do
+  version '1.1.1'
+  sha256 '7331e72cfff30661606387c86e8153b2526547b588525385ac2ecd272d38194a'
+
+  # edgesuite.net is the official download host per the vendor homepage
+  url 'http://esd.lego.com.edgesuite.net/digitaldelivery/mindstorms/6ecda7c2-1189-4816-b2dd-440e22d65814/public/LMS-EV3-OSX-ENUS-01-01-01-full-setup.dmg'
+  name 'lego-mindstorms-ev3'
```
Comment (vitorgalvao) [inline Casks/lego-mindstorms-ev3.rb]: `name` is the place where you set the name of this app with no restrictions. So in this case it should be `Lego Mindstorms EV3 Home Edition IDE` (or whatever you find appropriate).

File: Casks/lego-mindstorms-ev3.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+cask :v1 => 'lego-mindstorms-ev3' do
+  version '1.1.1'
+  sha256 '7331e72cfff30661606387c86e8153b2526547b588525385ac2ecd272d38194a'
+
+  # edgesuite.net is the official download host per the vendor homepage
```
Comment (vitorgalvao) [inline Casks/lego-mindstorms-ev3.rb]: I can’t confirm because the homepage is down. We’ll have to wait.

**Comment by vitorgalvao** [comment]
It’s missing `uninstall`. There are [scripts](https://github.com/caskroom/homebrew-cask/tree/master/developer/bin) and [documentation](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#uninstall-stanza-details) geared for that.

File: Casks/lego-mindstorms-ev3.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+cask :v1 => 'lego-mindstorms-ev3' do
```
Comment (vitorgalvao) [inline Casks/lego-mindstorms-ev3.rb]: This should be changed to `cask :v1 => 'lego-mindstorms' do`. The file’s name also should be changed to `lego-mindstorms.rb`.

**Comment by adityadalal924** [comment]
`.pkg` need an uninstall stanza. You can find details on how to write one [here](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#uninstall-stanza-details).

Let us know if you have any trouble!

File: Casks/lego-mindstorms.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  # edgesuite.net is the official download host per the vendor homepage
+  url 'http://esd.lego.com.edgesuite.net/digitaldelivery/mindstorms/6ecda7c2-1189-4816-b2dd-440e22d65814/public/LMS-EV3-OSX-ENUS-01-01-01-full-setup.dmg'
+  name 'Lego Mindstorms EV3 Home Edition'
+  homepage 'http://www.lego.com/en-us/mindstorms'
+  license :gratis
+
+  pkg 'LEGO MINDSTORMS EV3 Home Edition.pkg'
+  uninstall :pkgutil => 'com.ni.pkg.lego*'
+  uninstall :pkgutil => 'com.microsoft.SilverlightInstaller'
```
Comment (vitorgalvao) [inline Casks/lego-mindstorms.rb]: We can’t do this. If the user has Silverlight installed for other reason, we’ll be breaking it.

File: Casks/lego-mindstorms.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  sha256 '7331e72cfff30661606387c86e8153b2526547b588525385ac2ecd272d38194a'
+
+  # edgesuite.net is the official download host per the vendor homepage
+  url 'http://esd.lego.com.edgesuite.net/digitaldelivery/mindstorms/6ecda7c2-1189-4816-b2dd-440e22d65814/public/LMS-EV3-OSX-ENUS-01-01-01-full-setup.dmg'
+  name 'Lego Mindstorms EV3 Home Edition'
+  homepage 'http://www.lego.com/en-us/mindstorms'
+  license :gratis
+
+  pkg 'LEGO MINDSTORMS EV3 Home Edition.pkg'
+  uninstall :pkgutil => 'com.ni.pkg.lego*'
```
Comment (vitorgalvao) [inline Casks/lego-mindstorms.rb]: Please add an empty line above `uninstall`.

**Comment by vitorgalvao** [comment]
Please [squash your commits](http://davidwalsh.name/squash-commits-git).

**Comment by 2manyprojects2littletime** [comment]
Is there anything else to action from my end?

**Comment by adityadalal924** [comment]
Just took another look at the Cask, I'm thinking it should be `lego-mindstorms-ev3` (because there are other installers for lego mindstorms, and also, the uninstall stanza should be more specific (ie. list out the packages to uninstall, instead of using the wildcard), as it could break people's installs with other versions of the software installed.

**Comment by 2manyprojects2littletime** [comment]
```com.microsoft.SilverlightInstaller
com.ni.pkg.lego.ev3.Eng.1.1.1
com.ni.pkg.lego.x3.1.1.1.core
com.ni.pkg.lego.x3.1.1.1.update
com.ni.pkg.legodriver```
are the pkg ids it installs, should it specify all the *.lego* ones in the uninstall or just the one specific to ev3?

**Comment by adityadalal924** [comment]
Ah I see. In this case, do something like:

```
uninstall :pkgutil => [
                         'com.ni.pkg.lego.ev3.Eng.1.1.1',
                         'com.ni.pkg.lego.x3.1.1.1.core',
                         'com.ni.pkg.lego.x3.1.1.1.update'
                        ]

zap :pkgutil => 'com.ni.pkg.legodriver'
```

As of right now, there aren't other lego mindstorms casks, but explicitly defining things to uninstall is always a better option.

Comment when the Cask is updated (or if you have any problems) so that we get a notification and can go ahead and merge it.

Thanks!

**Comment by 2manyprojects2littletime** [comment]
I think that's everything, waiting on CI.

**Comment by adityadalal924** [comment]
Thank you for the contribution. It was merged directly as d0187946ed1b3e0579efe48a73af128a10beb15b to keep commit history cleaner. Your contribution is still credited to you.

**Comment by vitorgalvao** [comment]
@adityadalal924 Were you able to verify the download url? I still can’t access the website.

**Comment by adityadalal924** [comment]
@vitorgalvao Yep, it downloads fine on my end.

```
wget http://esd.lego.com.edgesuite.net/digitaldelivery/mindstorms/6ecda7c2-1189-4816-b2dd-440e22d65814/public/LMS-EV3-OSX-ENUS-01-01-01-full-setup.dmg
--2015-12-11 18:25:55--  http://esd.lego.com.edgesuite.net/digitaldelivery/mindstorms/6ecda7c2-1189-4816-b2dd-440e22d65814/public/LMS-EV3-OSX-ENUS-01-01-01-full-setup.dmg
Resolving esd.lego.com.edgesuite.net... 65.216.231.88, 65.216.231.121
Connecting to esd.lego.com.edgesuite.net|65.216.231.88|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 664126388 (633M) [application/octet-stream]
Saving to: 'LMS-EV3-OSX-ENUS-01-01-01-full-setup.dmg'
{snip}
```

**Comment by vitorgalvao** [comment]
I mean if you were able to verify that it is indeed the correct download as shown by the homepage (since hosts differ).

**Comment by adityadalal924** [comment]
Yes, the `url` is the same one as found on http://www.lego.com/en-us/mindstorms/downloads/download-software (http://esd.lego.com.edgesuite.net/digitaldelivery/mindstorms/6ecda7c2-1189-4816-b2dd-440e22d65814/public/LMS-EV3-OSX-ENUS-01-01-01-full-setup.dmg)

**Pull request closed without merging**